adaptor/objectives/distillation.py

```diff
@@ -0,0 +1,168 @@
+import abc
+import inspect
+from typing import Optional, Union, Dict, Any, Tuple
+
+import torch
+from torch.nn import CrossEntropyLoss, CosineEmbeddingLoss
+from torch.nn.functional import log_softmax, softmax
+from transformers import BatchEncoding, PreTrainedModel
+from transformers.utils import ModelOutput
+
+from adaptor.lang_module import LangModule
+from adaptor.objectives.objective_base import Objective
+
+
+class Distillation(Objective, abc.ABC):
+
+    def __init__(self, *args,
+                 distilled_model: PreTrainedModel,
+                 temperature: int = 1,
+                 logits_ce_loss_weight: int = 1,
+                 hidden_cossim_loss_weight: int = 1,
+                 add_hidden_states_loss: bool = False,
+                 restrict_loss_to_mask: bool = False,
+                 **kwargs):
+        self.teacher_model = distilled_model
+
+        self.temperature = temperature
+        self.logits_ce_loss_weight = logits_ce_loss_weight
+        self.hidden_cossim_loss_weight = hidden_cossim_loss_weight
+
+        self.restrict_loss_to_mask = restrict_loss_to_mask
+        self.add_hidden_states_loss = add_hidden_states_loss
+
+        if add_hidden_states_loss:
+            # in this case, we'll need the teacher to yield the hidden states in the output
+            self.teacher_model.config.output_hidden_states = True
+
+        super().__init__(*args, **kwargs)
```

Comment:
try with super(Distillation, self)?

Reply:
It seems that `super(Distillation, self)` is a support of Python<3.X, which we don't support anyway. I will replace `super(Distillation, self)` calls with simple `super()`.